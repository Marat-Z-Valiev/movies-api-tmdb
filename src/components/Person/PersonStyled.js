import styled from "styled-components";

const PersonStyled = styled.div`
	display: flex;
	flex-direction: row;

	@media (min-width: 320px) and (max-width: 425px) {
		flex-direction: column;
	}

	img {
		margin-left: 40px;
	}

	.info {
		text-align: left;
		margin-left: 10px;
		padding-left: 15px;

		@media (min-width: 320px) and (max-width: 425px) {
			text-align: center;
		}

		p {
			font-size: 1.2rem;

			@media (min-width: 320px) and (max-width: 425px) {
				padding: 10px;
			}
		}
	}

	.fas {
		margin-right: 10px;
	}
`;

export default PersonStyled;
