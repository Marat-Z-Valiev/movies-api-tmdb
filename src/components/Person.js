import React, {useState, useEffect} from "react";
import axios from "axios";
import PersonStyled from "./styles/PersonStyled";
import Results from "./Results";
import noImage from "../images/no-image-available.jpg";
import Spinner from "./Spinner";

const Person = ({match}) => {
	const personId = match.params.id;
	let [personInfo, setPersonInfo] = useState({});
	let [movieCredits, setMovieCredits] = useState([]);
	let [isLoading, setIsLoading] = useState(true);

	const getPersonInfo = async () => {
		await axios
			.get(
				`https://api.themoviedb.org/3/person/${personId}?api_key=f8efee7e451d2ca98ae50114ad74aeeb&language=en-US`
			)
			.then(response => {
				setPersonInfo((personInfo = response.data));
				setIsLoading((isLoading = false));
			})
			.catch(err => console.log(`this is error ${err}`));
	};

	const getMovieCreidts = async () => {
		axios
			.get(
				`https://api.themoviedb.org/3/person/${personId}/movie_credits?api_key=f8efee7e451d2ca98ae50114ad74aeeb&language=en-US`
			)
			.then(response => {
				setMovieCredits((movieCredits = response.data.cast));
			})
			.catch(err => console.log(`this is error ${err}`));
	};

	useEffect(() => {
		getPersonInfo();
	}, []);

	useEffect(() => {
		getMovieCreidts();
	}, []);

	const {profile_path, name, birthday, biography} = personInfo;

	return (
		<>
			{isLoading ? (
				<Spinner />
			) : (
				<>
					<h1 style={{textAlign: "center"}}>Person page</h1>
					<PersonStyled>
						<div>
							{profile_path == null ? (
								<img
									className="no-image"
									src={noImage}
									alt="no image available"
								/>
							) : (
								<img
									src={`https://image.tmdb.org/t/p/w342/${profile_path}`}
									alt={name}
								/>
							)}
						</div>
						<div className="info">
							<h2>{name}</h2>
							{birthday ? (
								<h2>
									<i
										style={{color: "#ffffff"}}
										className="fas fa-birthday-cake"
									></i>
									{birthday}
								</h2>
							) : (
								""
							)}
							<p>{biography}</p>
						</div>
					</PersonStyled>
					<Results results={movieCredits.slice(0, 6)}></Results>
				</>
			)}
		</>
	);
};

export default Person;
