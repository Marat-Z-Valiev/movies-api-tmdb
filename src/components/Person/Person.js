import React, {useEffect} from "react";
import {useDispatch, useSelector} from "react-redux";
import {fetchPerson, personSelector} from "../../store/slices/person";
import {
	fetchMovieCredits,
	movieCreditsSelector,
} from "../../store/slices/movieCredits";
import PropTypes from "prop-types";
import PersonStyled from "./PersonStyled";
import Results from "../Results/Results";
import noImage from "../../images/no-image-available.jpg";
import Spinner from "../Spinner/Spinner";
import Error from "../Error/Error";

const Person = ({match}) => {
	const personId = match.params.id;
	const dispatch = useDispatch();
	const {loading, person, hasErrors} = useSelector(personSelector);
	const {movieCredits} = useSelector(movieCreditsSelector);

	useEffect(() => {
		dispatch(fetchPerson(personId));
	}, [dispatch]);

	useEffect(() => {
		dispatch(fetchMovieCredits(personId));
	}, [dispatch]);

	const {profile_path, name, birthday, biography, gender} = person;

	return (
		<>
			{hasErrors ? <Error /> : ""}
			{loading ? (
				<Spinner />
			) : (
				<>
					<h1 style={{textAlign: "center"}}>
						{gender === 1 ? "Actress" : "Actor"} page
					</h1>
					<PersonStyled>
						<div>
							{profile_path == null ? (
								<img
									className="no-image"
									src={noImage}
									alt="no image available"
								/>
							) : (
								<img
									src={`https://image.tmdb.org/t/p/w342/${profile_path}`}
									alt=""
								/>
							)}
						</div>
						<div className="info">
							<h2>{name}</h2>
							{birthday ? (
								<h2>
									<i
										style={{color: "#ffffff"}}
										className="fas fa-birthday-cake"
									></i>
									{birthday}
								</h2>
							) : (
								""
							)}
							<p>{biography}</p>
						</div>
					</PersonStyled>
					<h2 style={{textAlign: "center", fontSize: "1.8rem"}}>
						Cast In Movies
					</h2>
					<Results results={movieCredits.slice(0, 6)}></Results>
				</>
			)}
		</>
	);
};

Person.propTypes = {
	match: PropTypes.object.isRequired,
};

export default Person;
