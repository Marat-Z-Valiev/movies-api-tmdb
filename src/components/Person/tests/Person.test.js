import React from "react";
import {shallow, configure} from "enzyme";
import toJSON from "enzyme-to-json";
import Adapter from "enzyme-adapter-react-16";
import Person from "../Person";

configure({adapter: new Adapter()});

describe("Person", () => {
	it("should render correctly", () => {
		const props = {
			match: {
				params: {
					id: 1,
				},
			},
		};
		const wrapper = shallow(<Person {...props} />);
		expect(toJSON(wrapper)).toMatchSnapshot();
	});
});
