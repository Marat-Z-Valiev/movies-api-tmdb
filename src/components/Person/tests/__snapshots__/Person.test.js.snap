// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Person should render correctly 1`] = `
<Fragment>
  <Spinner />
</Fragment>
`;
